app/api_clients/base.py

```diff
@@ -41,9 +27,29 @@ def __init__(self, message: str):
         super().__init__(f"{self.client_name} API client error: {message}")
 
 
+class JWTTokenAuth(httpx.Auth):
+    def auth_flow(self, request: httpx.Request) -> Generator[httpx.Request, httpx.Response, None]:
+        # NOTE: Needs to be re-generated for each request as it exipres after a certain time
+        jwt_token = get_api_modifier_jwt_token()
+
+        request.headers["Authorization"] = f"Bearer {jwt_token}"
+
+        yield request
+
+
+class ClusterSecretAuth(httpx.Auth):
```

Comment:
Again, even if the concept of ClusterSecret is something specific of OptScale I would prefer to be more specific here.